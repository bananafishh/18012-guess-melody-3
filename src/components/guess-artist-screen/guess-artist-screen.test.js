import React from 'react';
import renderer from 'react-test-renderer';

import GuessArtistScreen from './guess-artist-screen.jsx';

const question = {
  type: `artist`,
  song: {
    artist: `X Ambassadors`,
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
  },
  answers: [
    {
      artist: `X Ambassadors`,
      picture: `https://api.adorable.io/avatars/128/1`,
    },
    {
      artist: `Bastille`,
      picture: `https://api.adorable.io/avatars/128/2`,
    },
    {
      artist: `Imagine Dragons`,
      picture: `https://api.adorable.io/avatars/128/3`,
    },
  ],
};

it(`Компонент GuessArtistScreen рендерится корректно`, () => {
  const tree = renderer
    .create(<GuessArtistScreen
      question={question}
      onAnswer={() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
