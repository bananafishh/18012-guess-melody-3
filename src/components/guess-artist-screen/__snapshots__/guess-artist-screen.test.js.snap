// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент GuessArtistScreen рендерится корректно 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    />
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-0"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-0"
      />
      <label
        className="artist__name"
        htmlFor="answer-0"
      >
        <img
          alt="X Ambassadors"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128/1"
        />
        X Ambassadors
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-1"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="Bastille"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128/2"
        />
        Bastille
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-2"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="Imagine Dragons"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128/3"
        />
        Imagine Dragons
      </label>
    </div>
  </form>
</section>
`;
